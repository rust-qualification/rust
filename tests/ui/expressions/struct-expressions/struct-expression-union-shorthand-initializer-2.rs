// Struct Expressions: 787ef148-019a-4599-b0f2-e155512cab99

union MyUnion {
    x: i32,
    y: f64,
}

fn main() {
    let x: i32 = 5;
    let y: f64 = 5.2;
    let shorthand_union_two_fields = MyUnion { x, y };
    //~^ ERROR union expressions should have exactly one field
}