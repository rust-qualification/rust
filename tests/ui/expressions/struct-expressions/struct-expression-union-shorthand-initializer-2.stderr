error[E0784]: union expressions should have exactly one field
  --> $DIR/struct-expression-union-shorthand-initializer-2.rs:11:38
   |
LL |     let shorthand_union_two_fields = MyUnion { x, y };
   |                                      ^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0784`.
